tasks/02_user.yml

```diff
@@ -50,26 +50,52 @@
       echo $POTOS_NEW_PASS
   register: potos_new_user_password
   changed_when: false
-  when: potos_firstboot_local_user_add_with_sudo or potos_firstboot_local_user_add_without_sudo
+  when: potos_firstboot_local_user_add
   tags:
     - 'molecule-notest'
 
-- name: Add local user with sudo permission
+- name: Add local groups
+  ansible.builtin.group:
+    name: "{{ item }}"
+    system: true
+  loop: "{{ potos_firstboot_local_groups }}"
+  when:
+    - potos_firstboot_local_groups is defined
+    - potos_firstboot_local_groups is iterable
+    - potos_firstboot_local_groups is not string
+    - potos_firstboot_local_groups is not mapping
+  tags:
+    - 'molecule-notest'
+
+- name: Add local user with secondary group memberships
   ansible.builtin.user:
     name: "{{ potos_new_user_password.stdout_lines[0] }}"
     password: "{{ potos_new_user_password.stdout_lines[1] | password_hash('sha512') }}"
-    groups: sudo
+    groups: "{{ potos_firstboot_local_user_groups | join(',') }}"
     shell: "{{ potos_firstboot_local_user_add_shell }}"
-  when: potos_firstboot_local_user_add_with_sudo
+  when:
+    - potos_firstboot_local_user_add
+    - potos_firstboot_local_user_groups is defined
+    - potos_firstboot_local_user_groups is iterable
+    - potos_firstboot_local_user_groups is not string
+    - potos_firstboot_local_user_groups is not mapping
+    - (potos_firstboot_local_user_groups | length > 0)
   tags:
     - 'molecule-notest'
 
-- name: Add local user without sudo permission
+- name: Add local user without secondary group memberships
   ansible.builtin.user:
     name: "{{ potos_new_user_password.stdout_lines[0] }}"
     password: "{{ potos_new_user_password.stdout_lines[1] | password_hash('sha512') }}"
     shell: "{{ potos_firstboot_local_user_add_shell }}"
-  when: potos_firstboot_local_user_add_without_sudo
+  when: >
+    potos_firstboot_local_user_add and
+    (  potos_firstboot_local_user_groups is not defined
+    or potos_firstboot_local_user_groups is not iterable
+    or potos_firstboot_local_user_groups is string
+    or potos_firstboot_local_user_groups is mapping
+    or ( potos_firstboot_local_user_groups | length == 0 )
+    )
   tags:
     - 'molecule-notest'
```

Comment:
Wouldn't it be possible to combine this into one task? If no group is defined just omitting the value? Now it feels a bit redundant as everything except the group is identical.

Reply:
I confess: I was too lazy to test the behaviour when `groups:` receives an empty set. I'll improve.

Reply:
@fadnincx I don't know if this solution is much cleaner, but it wokrs and I have tested all kinds of settings for the `potos_firstboot_local_user_groups` variable and still the local user gets created.